src/utils/flow.js

```diff
@@ -36,14 +36,60 @@ export function immutableUpdate<T: Object>(object: T, ...rest: Object[]): T {
  * This function takes a string and returns either a valid TabSlug or null, this doesn't
  * throw an error so that any arbitrary string can be converted, e.g. from a URL.
  */
-export function toValidTabSlug(tabSlug: string): TabSlug | null {
-  switch (tabSlug) {
+export function toValidTabSlug(tabSlug: any): TabSlug | null {
+  const coercedTabSlug = (tabSlug: TabSlug);
+  switch (coercedTabSlug) {
     case 'calltree':
     case 'stack-chart':
     case 'marker-chart':
     case 'marker-table':
-      return tabSlug;
-    default:
+      return coercedTabSlug;
+    default: {
+      // The coerced type SHOULD be empty here. If in reality we get
+      // here, then it's not a valid transform type, so return null.
+      const unhandledCase: empty = coercedTabSlug; // eslint-disable-line no-unused-vars
       return null;
+    }
   }
 }
+
+/**
+ * This function will take an arbitrary string, and try to convert it to a valid
+ * TransformType.
+ */
+export function convertToTransformType(type: string): TransformType | null {
+  // Coerce this into a TransformType even if it's not one.
+  const coercedType = ((type: any): TransformType);
+  switch (coercedType) {
+    // Exhaustively check each TransformType. The default arm will assert that
+    // we have been exhaustive.
+    case 'merge-call-node':
+    case 'merge-function':
+    case 'focus-subtree':
+    case 'focus-function':
+    case 'collapse-resource':
+    case 'collapse-direct-recursion':
+    case 'drop-function':
+      return coercedType;
+    default: {
+      // The coerced type SHOULD be empty here. If in reality we get
+      // here, then it's not a valid transform type, so return null.
+      const unhandledCase: empty = coercedType; // eslint-disable-line no-unused-vars
```

Comment:
as above, you can just write `(coercedType: empty)` and get rid of the unused var.

Reply:
See above.